[IMP] ecommerce: update checkout page (#17526)
odoo/documentation: content/applications/websites/ecommerce/checkout.rst

Default add to cart behavior
~~~~~~~~~~~~~~~~~~~~~~~~~~~~

When clicking the :guilabel:`Add to cart` button, the customer remains on the product page.

If the product requires configuration (e.g., has variants or optional products), the configurator allows the customer to choose the desired options before adding the product to the cart.

> To proceed directly to the cart or checkout, customers can use the :guilabel:`Buy now` button.

Reviewer: I would be a bit more precise.

```suggestion
When clicking the :guilabel:`Add to cart` button, the customer remains on the product page by default.
If the selected product has linked :ref:`optional products
  <ecommerce/products/cross_upselling>` or :ref:`configured variants <ecommerce/categories_variants/product-variants>`, a :guilabel:`Configure` pop-up opens. The customer can then choose between :icon:`fa-shopping-cart` :guilabel:`Add to cart` (to
  stay on the product page) and :guilabel:`Go to Checkout` :icon:`fa-chevron-right` (to be
  redirected to the cart).
  To proceed directly to the cart or checkout, customers can use the :ref:`Buy Now
  <ecommerce/checkout/buy-now>`  button. 
```

Author: Thank you for your suggestion. I have made the updates accordingly.

> Default add to cart behavior

Reviewer 2: FYI, if the product has multiple packagings, we also show the product configurator on the shop page (it works similarly to configurable attributes) 🙂 It's unrelated to your change, but someone probably forgot to update the docs. Might be worth fixing as part of your PR.

Author: Got it, since it behaves like variants in the configurator, I’ve included it along with variants. Please check if it looks good.

~~~~~~~~~~~~~~~~~~~~~~~~~~~~

When clicking the :guilabel:`Add to cart` button, the customer remains on the product page by default.

> On the shop page, if the selected product is configurable (i.e., has variants), a :guilabel:`Configure` pop-up opens to let the customer choose the desired options.

Reviewer 2: Not necessarily true: a product without variants can still be configurable (if it has `dynamic` or `no_variant` attributes)
```suggestion
On the shop page, if the selected product is configurable (i.e. it has configurable attributes), a :guilabel:`Configure` pop-up opens to let the customer choose the desired options.
```

Author: OK, I've updated it.

> On the product page, variant selection is done directly on the page. However, if the product has linked :ref:`optional products <ecommerce/products/cross_upselling>`, a :guilabel:`Configure` pop-up opens.

Reviewer 2: > However, if the product has linked :ref:`optional products <ecommerce/products/cross_upselling>`, a :guilabel:`Configure` pop-up opens.

This is also true on the shop page

Author: Okay, I’ve generalized it. Please check.

Default add to cart behavior
~~~~~~~~~~~~~~~~~~~~~~~~~~~~

When clicking the :guilabel:`Add to cart` button, the customer remains on the product page by default.
If the selected product has linked :ref:`optional products <ecommerce/products/cross_upselling>`

> or :ref:`configured variants <ecommerce/categories_variants/product-variants>`,

Reviewer 2: Not necessarily true: on the shop page, the product configurator popup is indeed shown if the product is configurable (i.e. if some of its attributes can be configured), but on the product page, we don't show the popup for such products (unless the product has optional products of course).

Author: Thanks for the clarification! I’ve updated the section to distinguish between the shop page and product page behavior. Please let me know if it looks correct now.

> a :guilabel:`Configure` pop-up opens.The customer can then choose between :icon:`fa-shopping-cart` :guilabel:`Add to cart`

Reviewer: ```suggestion
a :guilabel:`Configure` pop-up opens. The customer can then choose between :icon:`fa-shopping-cart` :guilabel:`Add to cart`
```

Just one space missing

Author: Oops, missed that space 😅 it’s fixed now.

(to stay on the product page) and :guilabel:`Go to Checkout` :icon:`fa-chevron-right`
(to be redirected to the cart).

> To proceed directly to the cart or checkout, customers can use the :ref:`Buy Now <ecommerce/checkout/buy-now>` button.

Reviewer 2: I think this is already mentioned below, so no need to mention it again here :)

Author: Got it, I’ve removed it.

Default add to cart behavior
~~~~~~~~~~~~~~~~~~~~~~~~~~~~

When clicking the :guilabel:`Add to Cart` button, the customer remains on
the product page by default.

> - On the **shop page**: if the selected product is configurable (i.e., it

Reviewer 3: ```suggestion
- On the **e-commerce landing page** (`/shop`): if the selected product is configurable (i.e., it
```
(if you feel shop/product pages aren't clear enough)

  has configurable attributes and/or multiple packagings), a :guilabel:`Configure` pop-up opens to let
  the customer choose the desired options.

> - On the **product page**: variant/packagings selection is done directly on the page.

Reviewer 2: Nit:
```suggestion
- On the **product page**: attributes/packagings configuration is done directly on the page.
```

> - On the **product page**: attributes/packagings configuration is done directly on the page.

Reviewer 3: ```suggestion
- On a specific **product page**: attributes/packagings configuration is done directly on the page.
```
